ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/capturePoint/splitFractionCalculation/SplitFractionTools.java

```diff
@@ -14,20 +16,31 @@ public static double appendWeightDistribution(double desiredWeightDistribution,
 
    private static double combineTwoShifts(double desiredShift, double currentShift, double nominalShift)
    {
+      //clamping input values, incase we get negative values
+      desiredShift = clamp(desiredShift, 0.0, 1);
+      currentShift = clamp(currentShift, 0.0, 1);
+      nominalShift = clamp(nominalShift, 0.0, 1);
+
       if (currentShift == -1.0)
          return desiredShift;
 
+      //transfer = desired, default = nominal
+
       if (desiredShift > nominalShift)
       {
          double desiredPercentShiftForward = (desiredShift - nominalShift) / (1.0 - nominalShift);
          double desiredShiftForward = desiredPercentShiftForward * (1.0 - currentShift);
-         return currentShift + desiredShiftForward;
+
+         //clamping the output to stay within bounds
+         return clamp((currentShift + desiredShiftForward), 0.00001, 0.99999);
```

Comment:
I'd recommend making 0.00001 a static variable 
```private final static double minimumFraction = 1e-5;```
and then changing these to 
```clamp(a, minimumFraction, 1.0 - minimumFraction);```
for consistency and ease of change.